### ISSUE TITLE
Spaces in JS block incorrectly flagged as indentation error

### ISSUE THREAD

**Issue by paulschreiber** [squizlabs/PHP_CodeSniffer#515]
From https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/330

Sample code:
```php
<?php
class Foo {
	public static function bar( $atts ) {
		$script = "
      <script>
        DV.load('https://www.documentcloud.org/documents/$id.js', {
          width: '" . esc_js( $width ) . "',
          height: '" . esc_js( $height ) . "',
          foo: 'bar'
        });
      </script>
      <style>#DV-viewer-{$id}</style>";

		return $script;
	}
}
```

When I run PHPCS:
```
$ phpcs -p -s -v --standard=WordPress-VIP sample.php
```

I get this output:
```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 8 | ERROR | [x] Tabs must be used to indent lines; spaces are not
   |       |     allowed
   |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

If you delete either the width or the height line, the error goes away.

@westonruter says:
> there's probably some oddities around the parser where it switches from tabs,
> to a non-PHP block, then with spaces for indentation.

**Comment by aik099** [comment]
Sounds pretty much like indentation is detected based on new line presence ignoring fact, that it's a multi-line string in PHP.

**Comment by gsherwood** [comment]
This was an intentional change, but I decided to revert it in the DisallowTabIndent sniff but didn't do it in the DisallowSpaceIndent sniff. I don't see any good reason for them being different, so I will change this sniff to match.
